### PR TITLE
LibWeb: Finish Canvas Context2D transform functions

### PR DESCRIPTION
**Pull request by bplaat**
I have added the two missing functions in Canvas Context2D transform and added a test to test if the transforms work properly.

Also work towards #17993

### CODE REVIEW

File: Userland/Libraries/LibWeb/HTML/Canvas/CanvasTransform.h [hunk 1]
```diff
@@ -66,6 +76,24 @@ class CanvasTransform {
...
     }
 
+    // https://html.spec.whatwg.org/multipage/canvas.html#dom-context-2d-settransform-matrix
+    void set_transform(Geometry::DOMMatrix2DInit& init)
+    {
+        // 1. Let matrix be the result of creating a DOMMatrix from the 2D dictionary transform.
+        auto& realm = static_cast<IncludingClass&>(*this).realm();
+        auto maybe_matrix = Geometry::DOMMatrix::create_from_dom_matrix_2d_init(realm, init);
+        if (maybe_matrix.is_exception())
+            return;
```
Comment (awesomekling) [inline Userland/Libraries/LibWeb/HTML/Canvas/CanvasTransform.h:86]: I don't see anything in the spec about silently swallowing exceptions here. Are we not supposed to throw them?

Comment (bplaat) [reply Userland/Libraries/LibWeb/HTML/Canvas/CanvasTransform.h]: Good point

File: Userland/Libraries/LibWeb/HTML/Canvas/CanvasTransform.h [hunk 2]
```diff
@@ -66,6 +76,25 @@ class CanvasTransform {
...
 
+    // https://html.spec.whatwg.org/multipage/canvas.html#dom-context-2d-settransform-matrix
+    WebIDL::ExceptionOr<void> set_transform(Geometry::DOMMatrix2DInit& init)
+    {
+        // 1. Let matrix be the result of creating a DOMMatrix from the 2D dictionary transform.
+        auto& realm = static_cast<IncludingClass&>(*this).realm();
+        auto maybe_matrix = Geometry::DOMMatrix::create_from_dom_matrix_2d_init(realm, init);
+        if (maybe_matrix.is_exception())
+            return maybe_matrix.exception();
+        auto matrix = maybe_matrix.release_value();
```
Comment (ADKaster) [inline Userland/Libraries/LibWeb/HTML/Canvas/CanvasTransform.h:87]: can't this be TRY()'d ?

Comment (bplaat) [reply Userland/Libraries/LibWeb/HTML/Canvas/CanvasTransform.h]: Oh really nice! I didn't know that you could use TRY with ExceptionOr. Now that I know this I have added a fix for an older commit of my to also use TRY in DOMMatrix and DOMPointReadOnly.

**Review by ADKaster — APPROVED** [review]

**Pull request merged**